docs/configuration.md

```diff
@@ -0,0 +1,457 @@
+## Configuration
+Configuration parameters could be provided to Playkit JS upon instantiation of the player instance.
+```js
+var config = {
+  // Configuration here
+};
+var player = playkit.core.loadPlayer(config);
+```
+
+#### Configuration Structure
+```js
+{
+  logLevel: string,
+  type: string,
+  playback: PKPlaybackConfigObject,
+  sources: PKSourcesConfigObject,
+  metadata: PKMetadataConfigObject,
+  plugins: PKPluginsConfigObject
+}
+```
+
+#### Default Configuration Values
+```js
+var config = {
+    logLevel: "ERROR",
+    type: "Unknown",
+    sources: {
+        hls: [],
+        dash: [],
+        progressive: []
+    },
+    plugins: {},
+    metadata: {
+        description: "",
+        poster: ""
+    },
+    playback: {
+        audioLanguage: "",
+        textLanguage: "",
+        useNativeTextTrack: false,
+        volume: 1,
+        playsinline: true,
+        preload: "none",
+        autoplay: false,
+        allowMutedAutoPlay: true,
+        muted: false,
+        options: {
+            html5: {
+                hls: {},
+                dash: {}
+            }
+        },
+        preferNative: {
+            hls: false,
+            dash: false
+        },
+        streamPriority: [
+            {
+                engine: "html5",
+                format: "hls"
+            },
+            {
+                engine: "html5",
+                format: "dash"
+            },
+            {
+                engine: "html5",
+                format: "progressive"
+            }
+        ]
+    }
+};
+```
+## 
+>### config.logLevel
+>##### Type: `string`
+>##### Default: `"ERROR"`
+>##### Description: Defined the player log level.
+>Possible values: `"DEBUG", "INFO", "TIME", "WARN", "ERROR", "OFF"`
+## 
+>### config.type
+>##### Type: `string`
+>##### Default: `"Unknown"`
+>##### Description: Defined the media type.
+>Possible values: `"Vod", "Live", "Image", "Audio", "Unknown"`
+## 
+>### config.sources
+>##### Type: `PKSourcesConfig`
+>```js
+>{
+>  dash: Array<PKMediaSourceObject>
+>  hls: Array<PKMediaSourceObject>
+>  progressive: Array<PKMediaSourceObject>
+>}
+>```
+>>##### Type `PKMediaSourceObject`
+>>```js
+>>{
+>>  mimetype: string,
+>>  url: string,
+>>  id: string, // optional
+>>  bandwidth: number, // optional
+>>  width: number, // optional
+>>  height: number, // optional
+>>  drmData: Array<PKDrmDataObject> // optional
+>>}
+>>```
+>>##### Type `PKDrmDataObject`
+>>```js
+>>{
+>>  licenseUrl: string,
+>>  scheme: string,
+>>  certificate: string // optional
+>>}
+>>```
+>##### Default:
+>```js
+>{
+>  hls: [],
+>  dash: [],
+>  progressive: []
+>}
+>```
+>##### Description: Defined the optional sources for playback.
+>Should map media source extension type to its array of sources.
```

Comment:
The optional sources for playback should map the media source extension type to its array of sources.

if I understand this sentence correctly, please use my change.